### ISSUE TITLE
[Question] Customise the Captain URL

### ISSUE THREAD

**Issue by james-soak** [caprover/caprover#1144]
Hello,

My question is regarding the `captain.<hostname>` default URL - main whether this is customisable or whether it can be accessed on another port without having to have this default in place?

The use case is when using across multiple environments such as `test.clientserver.com` and `staging.clientserver.com`, the captain prefix prevents the use of Cloudflare as their SSL certificates are only set for a single level of wildcarding. This is fine for containers running on the system but this prevents us being able to put the captain management console behind Cloudflare access and the firewall rules. 

If it were possible to set this as a custom domain (such as `captain-test.clientserver.com`) this would solve the problem, equally being able to access it via a custom port could also allow for locking down behind firewall rules.

Many thanks in advance.

**Comment by githubsaturn** [comment]
TLDR: no, it's not currently possible.

I will make some basic changes to make it possible though. Note that captain subdomain is in many parts of the application (CLI, frontend and etc) and you might have to deal with some minor glitches. For example, probably `caprover serversetup` won't work, and you'll have to log in to the dashboard on web and manually update the root domain and click on enable SSL etc etc.


Definitely not recommended for prod environments, but feel free to give it a shot for testing environments. I'll post the instructions shortly.

**Comment by githubsaturn** [comment]
- Install CapRover 
- Create a file at `/captain/data/config-override.json` with this content:
```
{
 "captainSubDomain":"captain-test"
}
```
- Run `docker service update captain-captain --force`
- Log in to your instance using `http://ip:3000`
- Change the root domain to `clientserver.com`
- Enable SSL
- Enable Force SSL
- Use `caprover login` on CLI to log into your instance from your CLI

**Comment by alihalabyah** [comment]
I know this is a closed topic but for anyone facing a similar issue, after changing the `captainSubDomain` nginx starts breaking due to cert files not being found, you will need to generate new cert files using `certbot` and place the files under the right `letsencrypt` directory which is `/captain/data/letencrypt/etc/live`, also keep in the mind the symbolic link and the fact that the original cert files are in the archive not in the live directory.

**Comment by youfun** [comment]
> I know this is a closed topic but for anyone facing a similar issue, after changing the `captainSubDomain` nginx starts breaking due to cert files not being found, you will need to generate new cert files using `certbot` and place the files under the right `letsencrypt` directory which is `/captain/data/letencrypt/etc/live`, also keep in the mind the symbolic link and the fact that the original cert files are in the archive not in the live directory.我知道这是一个封闭的话题，但对于任何面临类似问题的人来说，在更改 `captainSubDomain` nginx后，由于找不到cert文件而开始中断，您将需要使用 `certbot` 生成新的cert文件，并将文件放置在正确的 `letsencrypt` 目录下，即 `/captain/data/letencrypt/etc/live` ，还请记住符号链接和事实，即原来的证书文件是在存档，而不是在现场目录。

how to use certbot
